qt/applications/workbench/workbench/app/mainwindow.py

```diff
@@ -401,6 +411,12 @@ def setup_default_layouts(self):
 
     # ----------------------- Events ---------------------------------
     def closeEvent(self, event):
+        # Check whether or not to save project
+        if not self.project.saved:
+            # Offer save
+            if self.project.offer_save(self):
```

Comment:
We need to check if cancel was pressed here and call `event.reject()` if it was and return.